core/src/foundations/ionic.vars.scss

```diff
@@ -0,0 +1,118 @@
+/**
+ * This is an auto-generated file, please do not change it directly.
+ * Ionic Design System
+ */
+
+$ionic-color-primary-10: var(--ionic-color-primary-10, #f7faff);
```

Comment:
If these variables are guaranteed to always be defined due to `ionic.root.scss` why we do we need to repeat the fallback values here?

Reply:
This is to ensure that the variables always have a value assigned to them. Developers should be importing a global stylesheet that includes the content of `ionic.root.scss` or assigning their own values to these variables. 

This approach is most helpful in cases where the variable assignment may fail or not be initially available at the time of render. 

It is better to have a fallback experience/value vs. a completely broken experience. 

If we decided to remove this in the future, it is pretty trivial to update the script that generates that format.